File: core/src/main/java/org/strykeforce/thirdcoast/telemetry/util/SquareSignalGenerator.java
package org.strykeforce.thirdcoast.telemetry.util;

class SquareSignalGenerator extends SignalGenerator {

  public SquareSignalGenerator(
      double frequency, double phase, double amplitude, double offset, double invert) {
    super(frequency, phase, amplitude, offset, invert);
  }

  @Override
  double getValue(double time) {
    double t = frequency * time + phase;
    double val = Math.signum(Math.sin(2.0 * Math.PI * t));
    return invert * amplitude * val + offset;
  }

  @Override
  public String toString() {
    return "Square at " + super.toString();
  }
}

File: telemetry/src/main/kotlin/org/strykeforce/thirdcoast/telemetry/util/SquareSignalGenerator.kt
package org.strykeforce.thirdcoast.telemetry.util

internal class SquareSignalGenerator(
  frequency: Double, phase: Double, amplitude: Double, offset: Double, invert: Double
) : SignalGenerator(frequency, phase, amplitude, offset, invert) {

  override fun getValue(time: Double): Double {
    val t = frequency * time + phase
    val `val` = Math.signum(Math.sin(2.0 * Math.PI * t))
    return invert * amplitude * `val` + offset
  }

  override fun toString(): String {
    return "Square at " + super.toString()
  }
}
